### ISSUE TITLE
Deprecate returning false and friends in favor of promises

### ISSUE THREAD

**Issue by dblock** [alexa-js/alexa-app#134]
The change in https://github.com/alexa-js/alexa-app/pull/133 was backwards compatible, but we may want a major version change that supports only promises and just using:

```
Promise.resolve(handlerResult).then(res.send).catch(handleError);
```

We can also remove the `return false` functionality and make either returning a Promise or calling the callback required in order to send the response (or just call response.send() explicitly).

**Comment by ajcrites** [comment]
@dblock do we still want to support callbacks as well or can we purely support promises? This will also apply to pre/post.

**Comment by dblock** [comment]
I think a deprecation message should happen first, then a major version bump and only supporting promises with good documentation is 👍 for me.

**Comment by ajcrites** [comment]
Do we want to have a separate issue for actually implementing the changes and keep this simply for adding a deprecation message and updating the documentation?

**Comment by dblock** [comment]
We want to have some pull requests :)

**Comment by dblock** [comment]
https://github.com/alexa-js/alexa-app/pull/167

**Comment by ajcrites** [comment]
Can this issue be closed?

**Comment by dblock** [comment]
Closed via https://github.com/alexa-js/alexa-app/pull/167. 

Also @ajcrites you have committer access don't you. Just do what you think is right and we can argue about it later ;)
